test/utils/IntegrationTest.sol

```diff
@@ -0,0 +1,415 @@
+///                    particular legal or regulatory significance.
+abstract contract IntegrationTest is HyperdriveTest {
+    using Lib for *;
+    using FixedPointMath for uint256;
+
+    // Fixed rate used to configure market.
+    uint256 internal constant FIXED_RATE = 0.05e18;
+
+    // ETH whale account.
+    address internal ETH_WHALE = 0x00000000219ab540356cBB839Cbe05303d7705Fa;
+
+    /// @dev Configuration for the Integration testing suite.
+    struct IntegrationConfig {
+        address[] whaleAccounts;
+        IERC20 token;
+        IERC20 baseToken;
+        uint256 shareTolerance;
+        uint256 minTransactionAmount;
+        uint256 positionDuration;
+        bool enableBaseDeposits;
+    }
+
+    IntegrationConfig internal config;
+    IHyperdrive.PoolDeployConfig internal poolConfig;
+    HyperdriveFactory factory;
+    address deployerCoordinator;
+    bool internal immutable isBaseETH;
+
+    /// @dev Constructor for the Integration testing suite.
+    /// @param _config The Integration configuration.
+    constructor(IntegrationConfig storage _config) {
+        config = _config;
+        isBaseETH = config.baseToken == IERC20(ETH);
+    }
+
+    /// @notice Forge setup function.
+    /// @dev Inherits from HyperdriveTest and deploys the
+    ///      Hyperdrive factory, deployer coordinator, and targets.
+    function setUp() public virtual override {
+        super.setUp();
+
+        // Fund accounts with ETH and token from whales.
+        vm.deal(alice, 100_000e18);
+        vm.deal(bob, 100_000e18);
+        address[] memory accounts = new address[](2);
+        accounts[0] = alice;
+        accounts[1] = bob;
+        for (uint256 i = 0; i < config.whaleAccounts.length; i++) {
+            fundAccounts(
+                address(hyperdrive),
+                config.token,
+                config.whaleAccounts[i],
+                accounts
+            );
+        }
+
+        // Deploy the Hyperdrive Factory contract.
+        deployFactory();
+
+        // Set the deployer coordinator address and add to the factory.
+        deployerCoordinator = deployCoordinator();
+        factory.addDeployerCoordinator(deployerCoordinator);
+
+        // Deploy all Hyperdrive targets using  deployer coordinator contract.
+        deployTargets(
+            bytes32(uint256(0xdeadbeef)), // Deployment Id
+            bytes32(uint256(0xdeadbabe)), // Deployment Salt
+            5_000e18, // Contribution
+            false // asBase
+        );
+
+        // Start recording event logs.
+        vm.recordLogs();
+    }
+
+    /// @dev Deploys all Hyperdrive contracts using the
+    ///      deployer coordinator contract.
+    /// @param deploymentId The deployment id.
+    /// @param deploymentSalt The deployment salt for create2.
+    /// @param contribution The amount to initialize the market.
+    /// @param asBase Initialize the market with base token.
+    function deployTargets(
```

Comment:
This isn't just deploying the targets. It's also deploying the pool. Why don't we call it `deploy` or `deployHyperdrive`?